### PR TITLE
feat: issue templates

### PR DESCRIPTION
**Pull request by aarnphm**
## What does this PR address?

For using new yaml syntax: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

## Who can help review?

cc @coffee-cup

For reference:
- the PULL_REQUEST_TEMPLATE: https://github.com/aarnphm/nixpacks/blob/feat/issue_template/.github/PULL_REQUEST_TEMPLATE.md
- feature request template: https://github.com/aarnphm/nixpacks/blob/feat/issue_template/.github/ISSUE_TEMPLATE/feature_request.yml
- bug report template: https://github.com/aarnphm/nixpacks/blob/feat/issue_template/.github/ISSUE_TEMPLATE/bug_report.yml

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<!--- If you plan to update documentation or tests in follow-up, please note -->
+
+- [ ] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
+- [ ] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
+- [ ] Can you build the project with `cargo build` and run the example with `cargo run --build examples/node --name node`?
+- [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
+- [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
+- [ ] Did you run `cargo check` and `cargo clippy` locally to ensure that CI pass?
+
+## Who can help review?
```
Comment (Milo123459) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: I'm not so sure on this section. I think most contributors and maintainers just check over PRs, and then can just be forwarded on.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask -->
+<!--- If you plan to update documentation or tests in follow-up, please note -->
+
+- [ ] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
+- [ ] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
+- [ ] Can you build the project with `cargo build` and run the example with `cargo run --build examples/node --name node`?
+- [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
+- [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
+- [ ] Did you run `cargo check` and `cargo clippy` locally to ensure that CI pass?
```
Comment (Milo123459) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: Should add `cargo fmt` here

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    description: |
+      Is there any way that you could help, e.g. by submitting a PR? Please make sure to read the [CONTRIBUTING.md](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md).
+    options:
+      - label: I can contribute code.
+        required: false
+      - label: I can contribute documentation.
+        required: false
+      - label: I can help testing the feature.
+        required: false
+      - label: I will be open to provide feedback once the feature is released.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:40]: Not sure this is really needed, either

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  attributes:
+    label: Contribution
+    description: |
+      Is there any way that you could help, e.g. by submitting a PR? Please make sure to read the [CONTRIBUTING.md](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md).
+    options:
+      - label: I can contribute code.
+        required: false
+      - label: I can contribute documentation.
+        required: false
+      - label: I can help testing the feature.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:38]: I don't think this is needed either.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      If this is related to another GitHub issue, please link here too.
+      If you have a current workaround, please also provide it here.
+    placeholder: |
+      This feature would solve issue ...
+- type: checkboxes
+  id: contribution
+  attributes:
+    label: Contribution
+    description: |
+      Is there any way that you could help, e.g. by submitting a PR? Please make sure to read the [CONTRIBUTING.md](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md).
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:32]:
```suggestion
      Is there any way that you could help, e.g. by submitting a PR? Please make sure to read the [contributing guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md).
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      An implementation of ... can be done as follow:
+- type: textarea
+  id: motivation
+  validations:
+    required: false
+  attributes:
+    label: Motivation
+    description: |
+      Please outline the motivation for the proposal. Is your feature request related to a problem? e.g., I'm always frustrated when [...].
+      If this is related to another GitHub issue, please link here too.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:23]:
```suggestion
      If this is related to another issue, please link here too.
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+- type: textarea
+  id: feature-request
+  validations:
+    required: true
+  attributes:
+    label: Feature request
+    description: |
+      A clear and concise description of the feature proposal.
+    placeholder: |
+      An implementation of ... can be done as follow:
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:14]:
```suggestion
      An implementation could be implemented by...
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+- type: textarea
+  id: motivation
+  validations:
+    required: false
+  attributes:
+    label: Motivation
+    description: |
+      Please outline the motivation for the proposal. Is your feature request related to a problem? e.g., I'm always frustrated when [...].
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:22]:
```suggestion
      Please outline the motivation for this feature request. Is your feature request related to a problem? e.g., I'm always frustrated when [...].
```

**Review by coffee-cup — APPROVED** [review]
This is awesome! Thank you 😄

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 9]
```diff
@@ -0,0 +1,54 @@
+name: 🐛 Bug Report
+description: Create a report to help us improve Nixpacks.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:2]:
```suggestion
description: Create a bug report to help us improve Nixpacks.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+title: "bug: "
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:3]:
```suggestion
title: ""
```
Title should be left blank IMO.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+labels: ["bug"]
+body:
+  - type: checkboxes
+    id: issue-already-exists
+    attributes:
+      label: Is there an existing issue for this?
+      description: |
+        Please search to see if an issue already exists for the bug you encountered.
+        Please see [Searching Issues and Pull Request](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how to use the GitHub search bar and filters.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:12]:
```suggestion
        Please see [Searching Issues and Pull Requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how to use the GitHub search bar and filters.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+      - label: I have searched the existing issues
+        required: true
+  - type: textarea
+    id: describe-the-bugs
+    validations:
+      required: true
+    attributes:
+      label: Describe the bug
+      description: Please provide a clear and concise description about the problem you ran into.
+      placeholder: I ran into ... while using ...
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:23]:
```suggestion
      placeholder: This happened when I...
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+  - type: textarea
+    id: to-reproduce
+    validations:
+      required: false
+    attributes:
+      label: To reproduce
+      description: |
+        Please provide a code sample or a code snipet to reproduce said problem. If you have code snippets, error messages, stack trace please also provide them here.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:31]:
```suggestion
        Please provide a code sample or a code snippet to reproduce the said problem. If you have code snippets, error messages or a stack trace please provide them here.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+        **IMPORTANT**: make sure to use [code tag](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) to correctly format your code. Screenshot is helpful but don't use it for code snippets as it doesn't allow others to copy-and-paste your code.
+      placeholder: |
+        Steps to reproduce the behaviour:
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:35]:
```suggestion
        Steps to reproduce the bug:
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      required: false
+    attributes:
+      label: Expected behavior
+      description: "A clear and concise description of what you would expect to happen."
+  - type: textarea
+    id: environment-info
+    attributes:
+      label: Environment
+      description: |
+        Please share your environment with us.
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/bug_report.yml:52]:
```suggestion
        Please share your environment with us. (ie, OS, OS version and Nixpacks version)
```

**Review by Milo123459 — CHANGES_REQUESTED** [review]

**Comment by Milo123459** [comment]
Sorry for lots of comments, just grammar things mainly!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+Fixes # (issue)
+
+## Before submitting:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask -->
+<!--- If you plan to update documentation or tests in follow-up, please note -->
+
+- [ ] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
+- [ ] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
+- [ ] Can you build the project with `cargo build` and run the example with `cargo run --build examples/node --name node`?
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: sounds good, will remove this.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 3 above)
Comment (aarnphm) [reply .github/ISSUE_TEMPLATE/feature_request.yml:40]: I will remove it.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+  - type: textarea
+    id: to-reproduce
+    validations:
+      required: false
+    attributes:
+      label: To reproduce
+      description: |
+        Please provide a code sample or a code snipet to reproduce said problem. If you have code snippets, error messages, stack trace please also provide them here.
+
+        **IMPORTANT**: make sure to use [code tag](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) to correctly format your code. Screenshot is helpful but don't use it for code snippets as it doesn't allow others to copy-and-paste your code.
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply .github/ISSUE_TEMPLATE/bug_report.yml:33]: noted.

**Comment by Milo123459** [comment]
> @Milo123459 may I ask for the discord links?

discord.gg/railway

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Milo123459) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: IMO, I'd say remove it.

**Review by Milo123459 — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (aarnphm) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: sounds good.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+## Before submitting:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask -->
+<!--- If you plan to update documentation or tests in follow-up, please note -->
+
+- [ ] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
+- [ ] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
+- [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
+- [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
+- [ ] Did you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI pass?
```
Comment (Milo123459) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: This should be `Have you...`

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (aarnphm) [reply .github/PULL_REQUEST_TEMPLATE.md]: updated.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 18 above)
Comment (aarnphm) [inline .github/PULL_REQUEST_TEMPLATE.md:16]:
```suggestion
- [ ] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI pass?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 19]
```diff
@@ -13,4 +13,4 @@ Fixes # (issue)
 - [ ] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
 - [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
 - [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
-- [ ] Did you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI pass?
+- [ ] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI pass?
```
Comment (Milo123459) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: Missed it (sorry!) but it should be `passes`, not `pass`

Comment (aarnphm) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: right, sorry for my bad grammar :(

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 18 above)
Comment (aarnphm) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Did you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI passes?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+## Before submitting:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask -->
+<!--- If you plan to update documentation or tests in follow-up, please note -->
+
+- [ ] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
+- [ ] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
+- [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
+- [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
+- [ ] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI pass?
```
Comment (aarnphm) [inline .github/PULL_REQUEST_TEMPLATE.md:16]:
```suggestion
- [ ] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI passes?
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 21]
```diff
@@ -0,0 +1,19 @@
+blank_issues_enabled: true
+version: 2.1
+contact_links:
+  - name: Nixpacks' documentation
+    url: https://nixpacks.com/
+    about: Documentation about Nixpacks. 
+  - name: Railway.
+    url: https://railway.app/
+    about: Railway.
+  - name: Nixpacks' issue tracker
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/config.yml:10]: We probably don't need this

File: .github/ISSUE_TEMPLATE/config.yml [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+    url: https://github.com/railwayapp/nixpacks/issues
+    about: For creating issues/PR ticket for Nixpacks' development.
+  - name: Railway's Discord
+    url: discord.gg/railway
+    about: Contact the Railway team for support via Discord.
+  - name: Railway Blog
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/config.yml:16]: Probably don't need this either

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+- type: textarea
+  id: feature-request
+  validations:
+    required: true
+  attributes:
+    label: Feature request
+    description: |
+      A clear and concise description of the feature request.
+    placeholder: |
+      An implementation could be implemented by...
```
Comment (Milo123459) [inline .github/ISSUE_TEMPLATE/feature_request.yml:14]:
```suggestion
      This could be implemented by...
```

**Review by Milo123459 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+    url: https://blog.railway.app/
+    about: To get the latest news/updates from the Railway team.
```
Comment (aarnphm) [inline .github/ISSUE_TEMPLATE/config.yml:18]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+contact_links:
+  - name: Nixpacks' documentation
+    url: https://nixpacks.com/
+    about: Documentation about Nixpacks. 
+  - name: Railway.
+    url: https://railway.app/
+    about: Railway.
+  - name: Nixpacks' issue tracker
+    url: https://github.com/railwayapp/nixpacks/issues
+    about: For creating issues/PR ticket for Nixpacks' development.
```
Comment (aarnphm) [inline .github/ISSUE_TEMPLATE/config.yml:12]:
```suggestion
```

**Comment by aarnphm** [comment]
I believe this is GTG now :)

**Pull request merged**